src/Avalonia.Base/Utilities/DeferredSetter.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive.Disposables;
+using System.Text;
+
+namespace Avalonia.Utilities
+{
+    /// <summary>
+    /// A utility class to enable deferring assignment until after property-changed notifications are sent.
+    /// </summary>
+    /// <typeparam name="TProperty">The type of the object that represents the property.</typeparam>
+    /// <typeparam name="TValue">The type of value with which to track the delayed assignment.</typeparam>
```

Comment:
> The type of value with which to track the delayed assignment.

I'm having trouble wrapping my head around this sentence. The type of value tracks the delayed assignment? In what way?

Why is it `object` in one case and `(object value, int priority)` in another? What has the priority got to do with the value? Again, how does this tuple track the delayed assignment?

Reply:
It's object in one case since the only piece of data needed to track the assignment to a `DirectProperty` is the value to set, but for `PriorityValue`s it needs to track the value and the priority passed in while setting.